src/Umbraco.Core/Cache/FastDictionaryCacheProvider.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Umbraco.Core.Composing;
+
+namespace Umbraco.Core.Cache
+{
+    /// <summary>
+    /// Implements a fast <see cref="IAppCache"/> on top of a concurrent dictionary.
+    /// </summary>
+    internal class FastDictionaryCacheProvider : IAppCache
```

Comment:
Not sure about this name either "Fast"... why is it so fast?

Reply:
Could not find a better name - it's an app cache that works on top of a dictionary, but using Lazy values so that actually creating the value does not happen inside a whole dictionary lock - so it's... better that the normal dictionary one... dunno what name to use, ideas?